lib/cocoapods/installer/analyzer.rb

```diff
@@ -302,7 +302,8 @@ def embedded_target_pod_targets_by_host(aggregate_target, embedded_aggregate_tar
             pod_target_names = Set.new(aggregate_target.pod_targets_for_build_configuration(configuration_name).map(&:name))
             embedded_pod_targets = embedded_aggregate_target.pod_targets_for_build_configuration(configuration_name).select do |pod_target|
               if !pod_target_names.include?(pod_target.name) &&
-                 aggregate_target.pod_targets.none? { |aggregate_pod_target| (pod_target.specs - aggregate_pod_target.specs).empty? }
+                 aggregate_target.pod_targets.none? { |aggregate_pod_target| (pod_target.specs - aggregate_pod_target.specs).empty? } &&
+                 (libraries_only || pod_target.build_as_dynamic?)
```

Comment:
there is a `build_as_static?` method I believe.

Reply:
Yes, there is. Shall I rewrite to `(libraries_only || !pod_target.build_as_static?)`?

Reply:
we could also add `.reject(&:build_as_static?).none? { ... }`

Reply:
Sorry, I don't understand as I'm not that firm with ruby, could you help me here?

What array do you think of calling `.reject` on?

My intention was to thin out the `select` on line 303 and select only `pod_target`s that include dynamic frameworks. Hence the `pod_target.build_as_dynamic?`.

Reply:
@mfiebig you can do

```ruby
aggregate_target.pod_targets.reject(&:build_as_static?).none? { |aggregate_pod_target| (pod_target.specs - aggregate_pod_target.specs).empty? }
```

Reply:
You can change it to what I suggested off-hand and re-run your test to ensure it passes :)

Reply:
It fails :/

I'm not sure it should succeed. The `aggregate_target.pod_targets.reject(&:build_as_static?)` looks at the pods of `aggregate_target`, but what we need to look at is wether `pod_target` itself is static or not. I think it doesn't matter how the `pod_targets` on `aggregate_target` relate to `pod_target`, or does it?

Reply:
Oh I see you are correct. I was mistaken. This is fine.

Reply:
Great, thanks or the feedback anyway! 👍